### ISSUE TITLE
Error handling for spawned process

### ISSUE THREAD

**Issue by brenzosa** [dzek69/nfc-poll-wrapper#1]
This was on Windows -- not requesting for Windows support but perhaps we could add some error handling (NOTE: it works fine on Ubuntu 20.04 even without libnfc-examples installed -- its just that we have mixed dev environments:

```
W20220415-09:34:04.479(8)? (STDERR) events.js:377
W20220415-09:34:04.480(8)? (STDERR)       throw er; // Unhandled 'error' event
W20220415-09:34:04.480(8)? (STDERR)       ^
W20220415-09:34:04.481(8)? (STDERR)
W20220415-09:34:04.481(8)? (STDERR) Error: spawn stdbuf ENOENT
W20220415-09:34:04.482(8)? (STDERR)     at Process.ChildProcess._handle.onexit (internal/child_process.js:274:19)
W20220415-09:34:04.482(8)? (STDERR)     at onErrorNT (internal/child_process.js:469:16)
W20220415-09:34:04.482(8)? (STDERR)     at processTicksAndRejections (internal/process/task_queues.js:82:21)
W20220415-09:34:04.483(8)? (STDERR) Emitted 'error' event on ChildProcess instance at:
W20220415-09:34:04.483(8)? (STDERR)     at Process.ChildProcess._handle.onexit (internal/child_process.js:280:12)
W20220415-09:34:04.484(8)? (STDERR)     at onErrorNT (internal/child_process.js:469:16)
W20220415-09:34:04.484(8)? (STDERR)     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
W20220415-09:34:04.484(8)? (STDERR)   errno: -4058,
W20220415-09:34:04.485(8)? (STDERR)   code: 'ENOENT',
W20220415-09:34:04.485(8)? (STDERR)   syscall: 'spawn stdbuf',
W20220415-09:34:04.485(8)? (STDERR)   path: 'stdbuf',
W20220415-09:34:04.486(8)? (STDERR)   spawnargs: [ '-o0', '-e0', 'nfc-poll', '-v' ]
W20220415-09:34:04.486(8)? (STDERR) }
=> Exited with code: 1
=> Your application is crashing. Waiting for file change.
```

This might work:

`        this._process.on("error", (err) => console.error("nfc-poll-error", err));`

**Comment by dzek69** [comment]
Hello, thanks for reporting this.

I missed that out, because when `stdbuf` is installed it won't trigger `error` event, even if `nfc-poll` is not (this is what I have on my environment right now) - it will just exit with 127 exit code.

I fixed that by catching that event and re-emitting as `process-error`. New version is now released to npm.

I also pushed some changes I left uncommited - event emitter is now typed so you will get extra benefits if you are using TypeScript.

Let me know if there is still something wrong for your setup.
